exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.java

```diff
@@ -44,8 +44,15 @@
  * that a certain key is mapped to a particular value <em>or</em> that there are no mapping for
  * the key in the map.
  *
- * <p>This map is implemented as a Merkle-Patricia tree. It does not permit null keys and values,
- * and requires that keys are 32-byte long.
+ * <p>This map is implemented as a Merkle-Patricia tree. It does not permit null keys and values.
+ *
+ * <p>This map can be instantiated in two ways - either with keys hashing with methods
+ * {@link #newInstance(String, View, Serializer, Serializer)} and
+ * {@link #newInGroupUnsafe(String, byte[], View, Serializer, Serializer)} or with no key hashing
+ * with methods {@link #newInstanceNoKeyHashing(String, View, Serializer, Serializer)} and
+ * {@link #newInGroupUnsafeNoKeyHashing(String, byte[], View, Serializer, Serializer)}. In case of
+ * no key hashing keys are required to be 32-byte long. Note that the former option is considered
```

Comment:
We discussed why it shall be the default option, and the documentation must properly reflect that.